jib-core/src/test/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilderTest.java

```diff
@@ -84,12 +85,33 @@ private static void verifyNextTarArchiveEntryIsDirectory(
     assertThat(extractionPathEntry.getMode()).isEqualTo(TarArchiveEntry.DEFAULT_DIR_MODE);
   }
 
-  private static FileEntry defaultLayerEntry(Path source, AbsoluteUnixPath destination) {
+  /**
+   * Verifies that the modification time has been reset in the PAX headers.
+   *
+   * @param entry The archive entry in the layer.
+   */
+  private static void verifyThatModificationTimeIsReset(ArchiveEntry entry) {
+    assertThat(entry.getLastModifiedDate().toInstant()).isEqualTo(Instant.EPOCH);
+  }
+
+  private static FileEntry layerEntry(Path source, AbsoluteUnixPath destination, FilePermissions permissions, String ownership) throws IOException {
     return new FileEntry(
         source,
         destination,
-        FileEntriesLayer.DEFAULT_FILE_PERMISSIONS_PROVIDER.get(source, destination),
-        FileEntriesLayer.DEFAULT_MODIFICATION_TIME);
+        permissions,
+        // Here we make sure to use the actual modification-time here because that's what would happen in
+        // regular use when copying the file from disk into the layer.
```

Comment:
Yes, I can see my comment was a bit unclear.  What I meant was that in normal use, Jib reads in files from disk and they have some "real" modification time.  Then Jib is supposed to reset this, but that didn't happen (or at least didn't happen "enough" because there were still PAX headers with the timestamps in them). 
Because the test helper created test files where the modtime was already reset, this issue wasn't visible in the test suite.

Reply:
> Then Jib is supposed to reset this, but that didn't happen (or at least didn't happen "enough" because there were still PAX headers with the timestamps in them).

This does happen. To clarify, the ordinary modification time is correctly set to epoch+1. It's just that the newer apache compress library added some new PAX headers. AFAICT, the mod time and the time values in PAX headers are independent. Everything works as designed with or without this CL, except that the new headers affect reproducibility. As long as we set the ordinary mod time to epoch+1, it doesn't matter what time you set for the PAX headers. So ideally, I think what we need to test is this:
- Verify that Jib continues to set the ordinary mod time to epoch+1 by default.
- Verify that Jib wipes out the time PAX headers. I think the value doesn't matter, as long as people don't complain.

Reply:
Ad 1: The PAX headers take precedence it seems, so I'll set them to 1 and try to switch the ordering like you suggested below 👍 
Ad 2: I believe that testing the wiping of the headers is captured in the existing test case `testToBlob_reproducibility`, as it compares the raw bytes.  This test would fail after I updated the helper function here, hence my point with the old version of this helper "hiding" the issue.

Reply:
If changing the order didn't work, then we should set the same modTime value to the PAX headers. Epoch+1 is just one possible value as a default, and modTime is user-configurable.

Reply:
Ah, yes of course, I forgot that it's user-configurable 👍

Reply:
Although - I'm a bit at a loss as to where the modTime is supposed to be reset.  In line 79 and 167 in `ReproducibleLayerBuilder.kt` there are calls to `.setModTime`.  In the former case, for directories, it's always set to EPOCH+1 regardless of any user settings.  In the latter case, for files, it's always set to the file's real value on disk, which is not reproducible.  Are there any other places in the code base where the modification time could've been reset?

Reply:
167 is the user-configured value. Just set the same time to all. For the (Jib-created) directiories, it is not customizable and always set to epoch+1.

To recap, just set the same value that Jib does it now.

Reply:
Hi @chanseokoh  - I've been away from this work for a while due to other things.  I've read the code more carefully now and I'll now understand that I was confused earlier as to where the responsibility of (re)setting the datetime lies.  I'll set the PAX headers to the same timestamp as setModTime 👍

Reply:
@bjornbugge too bad that you missed the release 3.4.1.